lib/reconfigure.js

```diff
@@ -94,6 +94,8 @@ module.exports = function(config) {
           outUrl = 'https://' + encodeURIComponent(credentials.host);
           if (credentials.apikey) {
             outCreds.iamApiKey = credentials.apikey;
+          } else if (credentials.url) {
```

Comment:
A "v2" `VCAP_SERVICES` `url` does not have embedded credentials, but the `host` `username` and `password` are present. I don't know if it is possible to construct a "v2" `VCAP_SERVICES` without an `apikey` - if it is then this isn't safe.
@emlaver thoughts?

Reply:
v2 VCAP, whether legacy or not, should have an `apikey`.  The non-legacy v2 VCAP has `host` and `username`, but does not have `url`  or `password`.  The legacy v2 VCAP has `host`, `username`, and `password`.
What we should do is replace the above with an `else if` block that adds username and password to url e.g.
```
...
} else if (credentials.username && credentials.password) {
  outUrl = 'https://' + encodeURIComponent(credentials.username) + ':' +
                     encodeURIComponent(credentials.password) + '@' + credentials.host;
}
```

Reply:
ok. I'll rewrite to always construct the URL from the components. The app I was looking at *did* have a `url`, so must have been an older version of VCAP